src/Index/Index.php

```diff
@@ -15,14 +15,31 @@ class Index implements ReadableIndex, \Serializable
     use EmitterTrait;
 
     /**
-     * An associative array that maps fully qualified symbol names to Definitions
+     * An associative array that maps namespaces to
+     *     an associative array that maps fully qualified symbol names
+     *     to global Definitions, e.g. :
+     *     [
+     *         'Psr\Log\LoggerInterface' => [
```

Comment:
Yes you're right it's a bit confusing. What about renaming the `namespaceDefinitions` to `fqnDefinitions` ? It's also confusing because there's already a
`getDefinition(string $fqn, bool $globalFallback = false)`
function where the `$fqn` parameter is `Psr\Log\LoggerInterface->log()`.